File: HotDesk_SPICE/Backup/Library/Library.Database/DAL/MenuNav.vb
Imports System.Data.SqlClient

Namespace DAL
    ''' <summary>
    ''' CapexCapital Data Access Layer
    ''' ------------------------------------------------
    ''' 15 March 2012  C.C.Yeon Initial Version
    ''' </summary>
    ''' <remarks></remarks>
    Public Class MenuNav
        Inherits Library.SQLServer.Connection

        Public Sub New()
            MyBase.New("SQLCon")
                  End Sub

                  ''20170809
                  Friend Function GetIDList() As DataTable
                           GetIDList = New Data.DataTable

                           With MyBase._cmd
                                    .CommandText = "SP_TMS_SEAT_DISPLAY"
                                    .CommandType = CommandType.StoredProcedure
                                    .CommandTimeout = 0
                           End With

                           MyBase._rdr = MyBase._cmd.ExecuteReader
                           GetIDList.Load(_rdr)
                  End Function
                  ''end 20170809

        Friend Function List(ByVal Table As String, ByVal SearchField As String, ByVal SearchValue As String, ByVal SortField As String, ByVal Direction As Integer, _
                             ByVal FromRowNo As Integer, ByVal ToRowNo As Integer, ByVal Deleted As Integer) As ListCollection
            List = New ListCollection

            Dim dt As New DataTable
            Dim ct As New DataTable
            Dim ds As New DataSet
          
            With MyBase._cmd
                .CommandText = "PSP_TESTING_LIST_CL"
                .CommandType = CommandType.StoredProcedure
                .CommandTimeout = 0

                .Parameters.Clear()

                .Parameters.Add(New SqlParameter("@Table", Table)).Direction = Data.ParameterDirection.Input
                .Parameters.Add(New SqlParameter("@Search", SearchField)).Direction = Data.ParameterDirection.Input
                .Parameters.Add(New SqlParameter("@Value", SearchValue)).Direction = Data.ParameterDirection.Input
                .Parameters.Add(New SqlParameter("@SortField", SortField)).Direction = Data.ParameterDirection.Input
                .Parameters.Add(New SqlParameter("@Direction", Direction)).Direction = Data.ParameterDirection.Input
                .Parameters.Add(New SqlParameter("@FrmRowno", FromRowNo)).Direction = Data.ParameterDirection.Input
                .Parameters.Add(New SqlParameter("@ToRowno", ToRowNo)).Direction = Data.ParameterDirection.Input
                .Parameters.Add(New SqlParameter("@Deleted", Deleted)).Direction = Data.ParameterDirection.Input
            End With

            MyBase._rdr = MyBase._cmd.ExecuteReader
            List.Data.Load(_rdr)

            While MyBase._rdr.Read
                List.TotalRow = _rdr("COUNTER")
            End While
        End Function

      


        Friend Function GetData(ByVal ID As String) As DataTable
            GetData = New Data.DataTable
           
            With MyBase._cmd
                .CommandText = "PSP_MM_RACK_WAREHOUSE_SEL"
                .CommandType = CommandType.StoredProcedure
                .CommandTimeout = 0

                .Parameters.Clear()
                .Parameters.Add(New SqlParameter("@ID", ID)).Direction = Data.ParameterDirection.Input
            End With

            MyBase._rdr = MyBase._cmd.ExecuteReader
            GetData.Load(_rdr)
        End Function

        Friend Function Maint(ByVal id As String, ByVal mods As String, ByVal RecType As String, ByVal UpdatedBy As String, ByVal UpdatedLoc As String, ByVal UpdatedCC As String) As String
            Maint = String.Empty

            With MyBase._cmd
                .CommandText = "PSP_TESTING_MAINT_CL"
                .CommandType = CommandType.StoredProcedure
                .CommandTimeout = 0

                .Parameters.Clear()
                .Parameters.Add(New SqlParameter("@ID", id)).Direction = Data.ParameterDirection.Input
                .Parameters.Add(New SqlParameter("@MOD", mods)).Direction = Data.ParameterDirection.Input

                .Parameters.Add(New SqlParameter("@RecType", RecType)).Direction = Data.ParameterDirection.Input
                .Parameters.Add(New SqlParameter("@CreatedBy", UpdatedBy)).Direction = Data.ParameterDirection.Input
                .Parameters.Add(New SqlParameter("@CreatedLoc", UpdatedLoc)).Direction = Data.ParameterDirection.Input
            End With
        
            MyBase._cmd.ExecuteNonQuery()
        End Function


                  Friend Function GetPalletData() As DataTable
                           GetPalletData = New Data.DataTable

                           With MyBase._cmd
                                    .CommandText = "SP_DDL_MPALLET"
                                    .CommandType = CommandType.StoredProcedure
                                    .CommandTimeout = 0
                           End With

                           MyBase._rdr = MyBase._cmd.ExecuteReader
                           GetPalletData.Load(_rdr)
                  End Function
    End Class

End Namespace

File: HotDesk_PRAI/Library/Library.Database/DAL/MenuNav.cs
using System.Data;
using System.Data.SqlClient;

namespace DAL
{
    /// <summary>
    /// CapexCapital Data Access Layer
    /// ------------------------------------------------
    /// 15 March 2012  C.C.Yeon Initial Version
    /// </summary>
    public class MenuNav : Library.SQLServer.Connection
    {
        public MenuNav() : base("SQLCon")
        {
        }

        // 20170809
        public DataTable GetIDList()
        {
            DataTable result = new DataTable();

            base._cmd.CommandText = "SP_TMS_SEAT_DISPLAY_PRAI";
            base._cmd.CommandType = CommandType.StoredProcedure;
            base._cmd.CommandTimeout = 0;

            base._rdr = base._cmd.ExecuteReader();
            result.Load(_rdr);

            return result;
        }
        // end 20170809

        public ListCollection List(string table, string searchField, string searchValue, string sortField, int direction, int fromRowNo, int toRowNo, int deleted)
        {
            ListCollection result = new ListCollection();

            var dt = new DataTable();
            var ct = new DataTable();
            var ds = new DataSet();

            base._cmd.CommandText = "PSP_TESTING_LIST_CL";
            base._cmd.CommandType = CommandType.StoredProcedure;
            base._cmd.CommandTimeout = 0;

            base._cmd.Parameters.Clear();
            base._cmd.Parameters.Add(new SqlParameter("@Table", table)).Direction = ParameterDirection.Input;
            base._cmd.Parameters.Add(new SqlParameter("@Search", searchField)).Direction = ParameterDirection.Input;
            base._cmd.Parameters.Add(new SqlParameter("@Value", searchValue)).Direction = ParameterDirection.Input;
            base._cmd.Parameters.Add(new SqlParameter("@SortField", sortField)).Direction = ParameterDirection.Input;
            base._cmd.Parameters.Add(new SqlParameter("@Direction", direction)).Direction = ParameterDirection.Input;
            base._cmd.Parameters.Add(new SqlParameter("@FrmRowno", fromRowNo)).Direction = ParameterDirection.Input;
            base._cmd.Parameters.Add(new SqlParameter("@ToRowno", toRowNo)).Direction = ParameterDirection.Input;
            base._cmd.Parameters.Add(new SqlParameter("@Deleted", deleted)).Direction = ParameterDirection.Input;

            base._rdr = base._cmd.ExecuteReader();
            result.Data.Load(_rdr);

            while (base._rdr.Read())
            {
                result.TotalRow = (int)_rdr["COUNTER"];
            }

            return result;
        }

        public DataTable GetData(string id)
        {
            DataTable result = new DataTable();

            base._cmd.CommandText = "PSP_MM_RACK_WAREHOUSE_SEL";
            base._cmd.CommandType = CommandType.StoredProcedure;
            base._cmd.CommandTimeout = 0;

            base._cmd.Parameters.Clear();
            base._cmd.Parameters.Add(new SqlParameter("@ID", id)).Direction = ParameterDirection.Input;

            base._rdr = base._cmd.ExecuteReader();
            result.Load(_rdr);

            return result;
        }

        public string Maint(string id, string mods, string recType, string updatedBy, string updatedLoc, string updatedCC)
        {
            string result = string.Empty;

            base._cmd.CommandText = "PSP_TESTING_MAINT_CL";
            base._cmd.CommandType = CommandType.StoredProcedure;
            base._cmd.CommandTimeout = 0;

            base._cmd.Parameters.Clear();
            base._cmd.Parameters.Add(new SqlParameter("@ID", id)).Direction = ParameterDirection.Input;
            base._cmd.Parameters.Add(new SqlParameter("@MOD", mods)).Direction = ParameterDirection.Input;
            base._cmd.Parameters.Add(new SqlParameter("@RecType", recType)).Direction = ParameterDirection.Input;
            base._cmd.Parameters.Add(new SqlParameter("@CreatedBy", updatedBy)).Direction = ParameterDirection.Input;
            base._cmd.Parameters.Add(new SqlParameter("@CreatedLoc", updatedLoc)).Direction = ParameterDirection.Input;

            base._cmd.ExecuteNonQuery();

            return result;
        }

        public DataTable GetPalletData()
        {
            DataTable result = new DataTable();

            base._cmd.CommandText = "SP_DDL_MPALLET";
            base._cmd.CommandType = CommandType.StoredProcedure;
            base._cmd.CommandTimeout = 0;

            base._rdr = base._cmd.ExecuteReader();
            result.Load(_rdr);

            return result;
        }
    }
}
